sympy/logic/boolalg.py

```diff
@@ -1274,14 +1343,52 @@ def _distribute(info):
             return info[0]
         rest = info[2](*[a for a in info[0].args if a is not conj])
         return info[1](*list(map(_distribute,
-            [(info[2](c, rest), info[1], info[2]) for c in conj.args])))
+            [(info[2](c, rest), info[1], info[2]) for c in conj.args])),\
+                remove_true=False)
     elif isinstance(info[0], info[1]):
         return info[1](*list(map(_distribute,
-            [(x, info[1], info[2]) for x in info[0].args])))
+            [(x, info[1], info[2]) for x in info[0].args])),\
+                remove_true=False)
     else:
         return info[0]
 
 
+def to_anf(expr, deep=True):
+    """
+    Converts expr to Algebraic Normal Form (ANF).
+
+    ANF is a canonical normal form, which means that two
+    equivalent formulas will convert to the same ANF.
+
+    A logical expression is in ANF if it has the form
+    ((A & B & ...) ^ (B & C ...) ^ B ^ True ^ ...),
+    i.e. it is purely true, purely false, conjunction of
+    variables or exclusive disjunction. The exclusive
+    disjunction can only contain true, variables or
+    conjunction of variables. No negations are permitted.
+
+    If ``deep`` if false, arguments of the boolean
+    expression considered variables, i.e. only the
+    top-level expression is converted to ANF.
+
+    Examples
+    ========
+    >>> from sympy.logic.boolalg import And, Or, Not, Implies, Equivalent
+    >>> from sympy.logic.boolalg import to_anf
+    >>> from sympy.abc import A, B, C
+    >>> to_anf(Not(A))
+    Xor(A, True)
+    >>> to_anf(And(Or(A, B), Not(C)))
+    Xor(A, B, A & B, A & C, B & C, A & B & C)
+    >>> to_anf(Implies(Not(A), Equivalent(B, C)), deep=False)
+    Xor(True, ~A, ~A & (Equivalent(B, C)))
+
+    """
+    if is_anf(expr):
+        return expr
+    return expr.to_anf(deep=deep)
```

Comment:
Is it possible that an un-sympified expression reaches here? Just like you sympified initially in `is_anf`.

Reply:
Done.

    expr = sympify(expr)

    if is_anf(expr):
        return expr
    return expr.to_anf(deep=deep)